src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -13,16 +13,26 @@
 using System.Threading.Tasks;
 using Microsoft.Diagnostics.Monitoring.EventPipe;
 using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tools.Common;
 using Microsoft.Internal.Common;
 using Microsoft.Internal.Common.Utils;
 
 namespace Microsoft.Diagnostics.Tools.Trace
 {
-    internal static class CollectCommandHandler
+    internal class CollectCommandHandler
     {
-        internal static bool IsQuiet { get; set; }
+        internal bool IsQuiet { get; set; }
 
-        private static void ConsoleWriteLine(string str)
+        public CollectCommandHandler()
+        {
+            Console = new DefaultConsole(false);
+            StartTraceSessionAsync = async (client, config, ct) => new CollectSession(await client.StartEventPipeSessionAsync(config, ct).ConfigureAwait(false));
+            ResumeRuntimeAsync = (client, ct) => client.ResumeRuntimeAsync(ct);
+            CollectSessionEventStream = name => new FileStream(name, FileMode.Create, FileAccess.Write);
+            IsOutputRedirected = Console.IsOutputRedirected;
```

Comment:
I don't think you need this property, the code can resume calling Console.IsOutputRedirected the same as it does from Console.IsInputRedirected.